redisinsight/api/src/modules/database-recommendation/scanner/strategies/search-JSON.strategy.ts

```diff
@@ -26,11 +14,8 @@ export class SearchJSONStrategy extends AbstractRecommendationStrategy {
   async isRecommendationReached(
     data: SearchJSON,
   ): Promise<IDatabaseRecommendationStrategyData> {
-  // todo: refactor. no need entire entity here
-    // TODO: [USER_CONTEXT] defer passing sessionMetadata to another PR
-    const { modules } = await this.databaseService.get({} as SessionMetadata, data.databaseId);
-
-    if (isRedisearchModule(modules)) {
+    // todo:improve decision mechanism when store Recommendations to avoid infinite checks when isReached:false
```

Comment:
did the `if (isRedisearchModule(modules))` prevent this scenario?

Reply:
no. it is not